pkg/providers/openstack/stage.go

```diff
@@ -54,20 +54,22 @@ func (p *OpenstackProvider) Stage(params types.StageImageParams) (_ *types.Image
 		return nil, errors.New("statting raw image file", err)
 	}
 
-	// TODO: Obtain image LOGICAL size, not actual (e.g. 10GB for OSv, not 8MB)
-	imageSizeB := rawImageFile.Size()
-	imageSizeMB := int(unikos.Bytes(imageSizeB).ToMegaBytes())
+	// Image size - use logical size unless physical size is bigger
+	imagePhysicalSizeMB := int(unikos.Bytes(rawImageFile.Size()).ToMegaBytes())
+	if imagePhysicalSizeMB > params.RawImage.RunSpec.MinInstanceDiskMB {
+		params.RawImage.RunSpec.MinInstanceDiskMB = int(imagePhysicalSizeMB)
+	}
 
 	// Pick flavor.
-	flavor, err := pickFlavor(clientNova, imageSizeMB, 0)
+	flavor, err := pickFlavor(clientNova, params.RawImage.RunSpec.MinInstanceDiskMB, 0)
```

Comment:
Do you think it would make sense to store the actual volume size defined in the flavor? It can actually be larger than the one we requested.